packages/date-picker/src/picker.vue

```diff
@@ -741,9 +741,12 @@ export default {
       this.picker.selectionMode = this.selectionMode;
       this.picker.unlinkPanels = this.unlinkPanels;
       this.picker.arrowControl = this.arrowControl || this.timeArrowControl || false;
-      if (this.format) {
-        this.picker.format = this.format;
-      }
+      this.$watch('format', (format) => {
+        // if main format is changed and pickerOptions.format undefined, then set to picker new main format
```

Comment:
pickerOptions.format is not documented (thus not publicly usable). I think you should always use main format.

Reply:
Hmmm, pickerOptions.format not publicly usable, but user can set this option. I have seen this several examples in various issues. Do you think it will be correctly: if reseting main format will be overpowering a special pickerOptions.format? I think there can be a situation, when main format mustn't be equal picker panel format. For example, main format for short view, and picker format for advanced view. I think it's better to be safe. However, in two projects, where I use the element, there was no such need :)
What do you think?

Reply:
Resetting is not a good idea, it can cause timing issues. Might be better to your implementation at this stage if we come to a consensus that pickerOption.format is an issue.

The reason that setting pickerOption.format can change panel's format is historical: Picker simply proxies all properties in pickerOptions to underlying panel without any sanity check. (But it shouldn't)

My thought is: pickerOption.format is not documented (undefined behavior / not expected to be used by the user), setting it should fail the component in perceivable ways. So users know they are doing something wrong.

Otherwise it might create a false sense of "it works". This can cause unnecessary inconveniences if we choose to refactor internal implementations in the future. (See https://github.com/ElemeFE/element/issues/9878)

@Leopoldthecoder What do you think?

Reply:
Agreed.